### PR TITLE
[WIP] feat(/functions): add remaining functions routes

### PR DESCRIPTION
**Pull request by i-pip**
## What kind of change does this PR introduce?
functions list action is already implemented, adding the remaining actions (get, post, patch,
delete). Beginning with this test for GET on /functions/:id

## What is the current behavior?
list method already implemented, missing get, post, patch & delete

Please link any relevant issues here.

#115 

## Additional context

Please pardon my linter's overzealous linting 🙈

### CODE REVIEW

**Comment by kiwicopple** [comment]
wow,  nice @i-pip 🔥 cool that you implemented this after the livestream.

@soedirgo does this look OK?

File: src/lib/PostgresMetaFunctions.ts [hunk 1]
```diff
@@ -20,4 +20,41 @@ export default class PostgresMetaFunctions {
...
+      const { data, error } = await this.query(sql)
+      if (error) {
+        return { data, error }
+      } else if (data.length === 0) {
+        return { data: null, error: { message: `Cannot find a function with ID ${id}` } }
+      } else {
+        return { data: data[0], error }
+      }
+    } else if (name) {
+      const sql = `${functionsSql} WHERE p.name = ${literal(name)};`
```
Comment (soedirgo) [inline src/lib/PostgresMetaFunctions.ts:44]: I think this should be `p.proname`?
```suggestion
      const sql = `${functionsSql} WHERE p.proname = ${literal(name)};`
```

**Review by soedirgo — CHANGES_REQUESTED** [review]
Thanks for the PR @i-pip! (And I like the lint!) Just one small fix :)

**Comment by i-pip** [comment]
> Thanks for the PR @i-pip! (And I like the lint!) Just one small fix :)

Thanks a ton @soedirgo! Fix applied 🕺🏾

**Pull request merged**